Server/ConcordiaCurriculumManager/Controllers/DossierController.cs

```diff
@@ -152,4 +152,37 @@ public async Task<ActionResult> CreateDossierForUser([FromBody, Required] Create
                 statusCode: StatusCodes.Status500InternalServerError);
         }
     }
+
+    [HttpDelete(nameof(DeleteDossier))]
+    [Authorize(Roles = RoleNames.Initiator)]
+    [SwaggerResponse(StatusCodes.Status400BadRequest, "Invalid input")]
+    [SwaggerResponse(StatusCodes.Status500InternalServerError, "Unexpected error")]
+    [SwaggerResponse(StatusCodes.Status201Created, "Dossier deleted successfully", typeof(DossierDTO))]
+    public async Task<ActionResult> DeleteDossier([FromBody, Required] DeleteDossierDTO dossier)
+    {
+        try
+        {
+            var user = await _userService.GetCurrentUser();
+            var deletedDossier = await _dossierService.DeleteDossier(dossier, user);
+            var deletedDossierDTO = _mapper.Map<DossierDTO>(deletedDossier);
+
+            return Created($"/{nameof(DeleteDossier)}", deletedDossierDTO);

```

Comment:
Normally an HTTP Delete returns a 204 (No Content) status code if the action has been enacted

Reply:
Yes, you are right. Should be good now